Components/LearningCore/Source/gov/sandia/cognition/statistics/StreamingSummaryStatistics.java

```diff
@@ -0,0 +1,222 @@
+    private double max;
+
+    /**
+     * The arithmetic mean of the values seen thus far
+     */
+    private double mean;
+
+    /**
+     * The intermediate value which is optimized for adding new values or
+     * returning actual variance
+     */
+    private double intermediateForVariance;
+
+    /**
+     * Initializes an empty summary. Values returned from this before any data
+     * is entered will not be good (max values, NaNs, etc.)
+     */
+    public StreamingSummaryStatistics()
+    {
+        numEntries = 0;
+        min = Double.MAX_VALUE;
+        max = -Double.MIN_VALUE;
+        mean = Double.NaN;
+        intermediateForVariance = Double.NaN;
+    }
+
+    /**
+     * Add the input value to the summary statistics
+     *
+     * @param x the value to add to the summary statistics
+     */
+    @PublicationReference(type = PublicationType.WebPage, title
+        = "Algorithms for calculating variance - Online algorithm", year = 2016,
+        url
+        = "https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Online_algorithm",
+        author = "Wikipedia")
+    @Override
+    public void addValue(double x)
+    {
+        if (Double.isNaN(mean))
+        {
+            mean = 0;
+            intermediateForVariance = 0;
+        }
+        min = Math.min(x, min);
+        max = Math.max(x, max);
+        ++numEntries;
+        double delta = x - mean;
+        mean += delta / numEntries;
+        intermediateForVariance += delta * (x - mean);
+    }
+
+    /**
+     * Merges the summary statistics stored in the input into this
+     *
+     * @param stats The other statistics to merge into this
+     */
+    @PublicationReference(type = PublicationType.WebPage, title
+        = "An average of standard deviations", year = 2010, author
+        = "See response by Dragon on 11/06/2010", url
+        = "http://www.talkstats.com/showthread.php/14523-An-average-of-standard-deviations")
+    @Override
+    public void merge(StreamingSummaryStatistics stats)
+    {
+        // No need to merge anything in if nothing is there
+        if (stats.numEntries == 0)
+        {
+            return;
+        }
+        // If I have nothing, I just copy him
+        if (numEntries == 0)
+        {
+            numEntries = stats.numEntries;
+            min = stats.min;
+            max = stats.max;
+            mean = stats.mean;
+            intermediateForVariance = stats.intermediateForVariance;
+            return;
+        }
+        // Now for the default case
+        // Updating the variance is a pain, and the equation I found for it requires
+        // the actual variance.  We'll back out to the intermediate in a moment
+        double sx2 = variance();
+        double sy2 = stats.variance();
+        double nx = numEntries();
+        double ny = stats.numEntries();
+        double xbar = mean();
+        double ybar = stats.mean();
+        double newVariance = ((sqr(nx) * sx2) + (sqr(ny) * sy2) - (ny * sx2)
```

Comment:
There are simpler ways of doing this. I have an implementation that I can provide for it.